File: examples/typescript/basic-generation.ts
/**
 * Basic text generation example using the AHF AI framework (TypeScript).
 *
 * Run: npx tsx examples/typescript/basic-generation.ts
 */

import { TextGenerator, Config } from "../../typescript/src/index.js";

async function main(): Promise<void> {
  const config = Config.fromObject({
    provider: "openai",
    apiKey: process.env["AHF_API_KEY"] ?? "sk-your-key-here",
    model: "gpt-4o",
    temperature: 0.7,
    maxTokens: 512,
    safetyLevel: "medium",
  });

  const generator = new TextGenerator(config);

  // Full generation
  const result = await generator.generate({
    prompt: "Explain the importance of open-source AI safety tools in three sentences.",
    systemPrompt: "You are a concise technical writer.",
  });

  console.log("Generated text:", result.text);
  console.log("Tokens used:", result.usage.totalTokens);
  console.log("Estimated cost: $" + result.usage.estimatedCost.toFixed(6));

  // Streaming generation
  console.log("\nStreaming output:");
  for await (const chunk of generator.stream({ prompt: "Count from 1 to 5." })) {
    process.stdout.write(chunk);
  }
  console.log();
}

main().catch(console.error);

File: examples/python/basic_generation.py
"""Basic text generation example using the AHF AI framework.

Demonstrates how to configure a provider and generate text
with safety filters enabled.
"""

import asyncio
from ahf_ai import TextGenerator, Config

async def main() -> None:
    # Create a configuration pointing at the desired provider.
    # In production, use Config.from_env() to read AHF_API_KEY, etc.
    config = Config(
        provider="openai",
        api_key="sk-your-api-key-here",
        model="gpt-4o",
        temperature=0.7,
        max_tokens=512,
        safety_level="medium",
    )

    # Instantiate the generator and produce text
    generator = TextGenerator(config)
    result = await generator.generate(
        prompt="Explain the importance of open-source AI safety tools in three sentences.",
        system_prompt="You are a concise technical writer.",
    )

    print(f"Generated text:\n{result.text}")
    print(f"\nTokens used: {result.usage.total_tokens}")
    print(f"Estimated cost: ${result.usage.estimated_cost:.6f}")
    print(f"Safety filtered: {result.safety_filtered}")

if __name__ == "__main__":
    asyncio.run(main())
